### DISCUSSION TITLE
获取oidc配置错误

### DISCUSSION THREAD

**Discussion by ningpengtao-coder** [alibaba/higress#1326] [General]
版本：higress-2.0.0
问题：
请求是出现这个错误以下错误：
[oidc] Failed to obtain OpenID configuration, current OIDC plugin is not working properly

希望能够有更多的日志，这样的报错不知道怎么处理。

**Comment by CH3CHO** [comment]
应该是你的插件配置有问题。建议提供配置信息以供分析。

**Reply by ningpengtao-coder** [reply]
> 应该是你的插件配置有问题。建议提供配置信息以供分析。

配置是这样的。

client_id: "test-nginx"
client_secret: "PW05jAe6hjBxgMl1OufOtCcFhMtaCTDx"
cookie_secret: "nqavJrGvRmQxWwGNptLdyUVKcBNZ2b18Guc1n_8DCfY="
oidc_issuer_url: "https://127.0.0.1:8443/auth/realms/intenral/"
redirect_url: "http://foo.bar.com/nacos/oauth2/callback/"
scope: "openid email offline_access"
service_name: "keycloak.static"
service_port: 80
service_source: "ip"

**Comment by ningpengtao-coder** [comment]
我希望能够有更多内部细节暴露，不然改配置比编码还痛苦。有什么办法提供日志等级吗？

**Reply by CH3CHO** [reply]
你可以试一下在 gateway pod 的命令行参数里加一个`--component-log-level wasm:debug`，启用 wasm 的 debug 日志。

**Comment by ningpengtao-coder** [comment]
@CH3CHO OAuthProxy.SetVerifier一直循环调用“{issuerURL}/.well-known/openid-configuration”,确定{issuerURL/.well-known/openid-configuration}是可以访问的。

**Reply by ningpengtao-coder** [reply]
[
![E552A7B8-9D90-4ba8-8E9F-ED8F7CCB8098](https://github.com/user-attachments/assets/01547503-fcf2-474a-ac9c-37647c8e1708)
](url)

**Reply by ningpengtao-coder** [reply]
provider/providers.go 73行：
pv, _ := internaloidc.NewProviderVerifier(context.TODO(), verifierOptions, providerJson)
你们修改的oauth2_proxy这里没有处理异常。

**Reply by johnlanni** [reply]
可以用一下最新的 oidc 插件，还有问题的话，建议提个issue方便我们跟踪
